pkg/ipamd/ipamd.go

```diff
@@ -2346,3 +2346,13 @@ func (c *IPAMContext) AddFeatureToCNINode(ctx context.Context, featureName rcv1a
 	newCNINode.Spec.Features = append(newCNINode.Spec.Features, newFeature)
 	return c.k8sClient.Patch(ctx, newCNINode, client.MergeFromWithOptions(cniNode, client.MergeFromWithOptimisticLock{}))
 }
+
+func (c *IPAMContext) tryAssignPodENI(ctx context.Context, pod *corev1.Pod, fnName string) error {
```

Comment:
Why do we need this ?

Reply:
Using this twice in the tests as a helper func:

https://github.com/dshehbaj/amazon-vpc-cni-k8s/blob/shehbaj/addipcnt-gauge-bug/pkg/ipamd/ipamd_test.go#L2437

https://github.com/dshehbaj/amazon-vpc-cni-k8s/blob/shehbaj/addipcnt-gauge-bug/pkg/ipamd/ipamd_test.go#L2447

I can move this helper func into the test module if we do not want to keep this in the main package.

Reply:
I moved this function to the test package now:

https://github.com/dshehbaj/amazon-vpc-cni-k8s/blob/shehbaj/addipcnt-gauge-bug/pkg/ipamd/ipamd_test.go#L2462